### ISSUE TITLE
Different problems on setup.sh script

### ISSUE THREAD

**Issue by arnaugamez** [malrev/ABD#2]
**EDIT: Please see comment below.**

Going through the materials and found a problem in hands-on1 to run _o-llvm.sh_ as no clang binary is found under /obfuscator/build/bin/ folder, where there is only _llvm-lit_

Environment:
- Fresh install of Ubuntu 18.04.3 on a VM
- Run ./setup.sh ./ as described in repository readme

Maybe some other dependency missing so setup.sh script not complete?

**Comment by arnaugamez** [comment]
I found the problem and it looks like it is on the setup script, as the _make -j$((`nproc` / 2))_ line will fail due to it will resolve to _make -j0_, which is invalid, in many default vbox installs (running only 1 core). Thus installation script will be aborted in that line and won't continue, resulting not only on clang missing, but also everything following from there.

A part from that, the setup.sh script has some other issues when running it into a fresh install of a OS like the one provided in the README by osboxes. Namely:

- The just described problem with make.
+This could be solved by enforcing a minimum of 1 core

- For installation of pip, it tries to get it through curl, which is not installed by default (would be installed afterwards during docker installation, but crashes before it as it does not have it yet). Also python3-distutils and python3-testresources are needed for the pip installation to succeed correctly without crashing or throwing errors.
+This could be solved by just installing those needed packages before proceeding with pip installation

- The first two commands on tigress part that add the repo sources and install docker are super redundant, as it has been installed just above when docker installation.
+Just remove those lines

- In general, there is a mix between using apt and apt-get.
+It should not be problematic but I would unify under apt.

I think I have solved all of them. Testing now to install from scratch with my modified setup.sh script and will make PR with it as soon I confirm it installs everything fine.
